src/lava/magma/core/model/py/ports.py

```diff
@@ -9,96 +9,229 @@
+
+    Class attributes
+    ----------------
+    VEC_DENSE : PyInPortVectorDense, default=None
+        Type of PyInPort.
+
+    VEC_SPARSE : PyInPortVectorSparse, default=None
+        Type of PyInPort.
+
+    SCALAR_DENSE : PyInPortScalarDense, default=None
+        Type of PyInPort.
+
+    SCALAR_SPARSE : PyInPortScalarSparse, default=None
+        Type of PyInPort.
     """
 
     VEC_DENSE: ty.Type["PyInPortVectorDense"] = None
     VEC_SPARSE: ty.Type["PyInPortVectorSparse"] = None
     SCALAR_DENSE: ty.Type["PyInPortScalarDense"] = None
     SCALAR_SPARSE: ty.Type["PyInPortScalarSparse"] = None
 
-    def __init__(self, csp_recv_ports: ty.List[CspRecvPort], *args):
-        self._csp_recv_ports = csp_recv_ports
-        super().__init__(*args)
-
-    @property
-    def csp_ports(self) -> ty.List[AbstractCspPort]:
-        """Returns all csp ports of the port."""
-        return self._csp_recv_ports
-
     @abstractmethod
     def recv(self):
+        """Abstract method to receive data (vectors/scalars) sent from connected
+        out Ports (source Ports). Removes the retrieved data from the channel.
+        Expects data on the channel and will block if there is no data to
+        retrieve on the channel.
+
+        Returns
+        -------
+        The point-wise added vectors or scalars received from connected Ports.
+        """
         pass
 
     @abstractmethod
     def peek(self):
+        """Abstract method to receive data (vectors/scalars) sent from connected
+        out Ports (source Ports). Keeps the data on the channel.
+
+        Returns
+        -------
+        The point-wise added vectors or scalars received from connected Ports.
+        """
         pass
 
     def probe(self) -> bool:
-        """Executes probe method of all csp ports and accumulates the returned
-        bool values with AND operation. The accumulator acc is initialized to
-        True.
+        """Method to check (probe) if there is data (vectors/scalars)
+        to receive from connected out Ports (source Ports).
+
+        Returns
+        -------
+        result : bool
+             Returns True only when probe returns True for all connected OutPort
+             channels.
 
-        Returns the accumulated bool value.
         """
-        # Returns True only when probe returns True for all _csp_recv_ports.
         return ft.reduce(
             lambda acc, csp_port: acc and csp_port.probe(),
-            self._csp_recv_ports,
+            self.csp_ports,
             True,
         )
 
 
 class PyInPortVectorDense(PyInPort):
+    """Python implementation of PyInPort for dense vector data."""
     def recv(self) -> np.ndarray:
+        """Method to receive data (vectors/scalars) sent from connected
+        out Ports (source Ports). Removes the retrieved data from the channel.
+        Expects data on the channel and will block if there is no data to
+        retrieve on the channel.
+
```

Comment:
One general comment: Sometimes, I see the term "out Ports" or similar. Sometimes, it is "OutPorts". I think we should be consistent.

Reply:
Changed all "out p/Ports" to "OutPorts"